From: Stuart Hayes <stuart_hayes@dell.com>
Subject: USB: fix error messages when USB keyboard is unplugged (#167170)
Patch-mainline: 2.6.17-rc1

This has been fixed in a much more complex and correct way in mainline,
but this simple fix will do what is needed for our kernel.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

 drivers/usb/input/hid-core.c |    1 +
 1 files changed, 1 insertion(+)

--- linux-2.6.16.orig/drivers/usb/input/hid-core.c
+++ linux-2.6.16/drivers/usb/input/hid-core.c
@@ -933,6 +933,7 @@ static void hid_irq_in(struct urb *urb, 
 		case -ESHUTDOWN:	/* unplug */
 		case -EILSEQ:		/* unplug timeout on uhci */
 			return;
+		case -EPROTO:		/* failed urbs unplug 1.1 dev from 2.0 hub */
 		case -ETIMEDOUT:	/* NAK */
 			break;
 		default:		/* error */
